dashboard/app/models/script_level.rb

```diff
@@ -372,6 +372,48 @@ def summarize_as_bonus
     }.camelize_keys
   end
 
+  def summarize_for_teacher_panel(student)
+    contained_levels = levels.map(&:contained_levels).flatten
+    if contained_levels.any?
+      user_level = student.last_attempt_for_any(contained_levels)
+      contained = true
+    else
+      user_level = student.last_attempt_for_any(levels)
+      contained = false
+    end
+
+    status = activity_css_class(user_level)
+    passed = [SharedConstants::LEVEL_STATUS.passed, SharedConstants::LEVEL_STATUS.perfect].include?(status)
+
+    if user_level
+      paired = user_level.paired?
+      if UserLevel.most_recent_driver(@script, @level, student)
+        driver = UserLevel.most_recent_driver(@script, @level, student)[0] # the name is in array index 0
+      end
+      if UserLevel.most_recent_navigator(@script, @level, student)
+        navigator = UserLevel.most_recent_navigator(@script, @level, student)[0] # the name is in array index 0
```

Comment:
`UserLevel.most_recent_navigator` hits the database too -- let's not call it twice.